src/map/settings.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * @packageDocumentation
+ * @module settings
+ */
+
+import assert from 'assert';
+
+import {HTMLSetting, Validator} from '../utils';
+
+export interface AxisPresets {    
+    /// Which property should we use for this axis
+    property?: string;
+    /// Which scale (linear/log) should we use
+    scale?: string;
+    /// The minimal value for this axis
+    min?: number;
+    /// The maximal value for this axis
+    max?: number;
+}
+
+const AXIS_DEFAULTS = {
+     property : '',
+     scale : 'linear',
+     min : 0,
+     max: 0,
+}
+
+/** HTML element holding settings for a given axis (x, y, z, color) */
+export class AxisSettings {
+    public property!: HTMLSetting<'string'>;
+    public scale!: HTMLSetting<'string'>;
+    public min!: HTMLSetting<'number'>;
+    public max!: HTMLSetting<'number'>;
+
+    constructor(properties: string[], presets?: AxisPresets) {
+        
+        // creates settings
+        this.max = new HTMLSetting('number', AXIS_DEFAULTS.max);
+        this.min = new HTMLSetting('number', AXIS_DEFAULTS.min);
+        this.property = new HTMLSetting('string', AXIS_DEFAULTS.property);
+        this.scale = new HTMLSetting('string', AXIS_DEFAULTS.scale);
+
+        this.property.validate = Validator<string>(properties.concat(['']), 'axis');
+        this.scale.validate = Validator<string>(['linear', 'log'], 'axis scale');
+        
+        this.applyPresets(presets);        
+    }
+        
+    public applyPresets(presets?: AxisPresets) : void {
+        /**
+         * Applies presets, possibly filling in with default values
+         */
+                 
+        /// Combine defaults with the given presets    
+        const initial : Required<AxisPresets> = { ...AXIS_DEFAULTS, ...presets };
```

Comment:
This would remove all customization made by the user when applying presets. Is it expected?